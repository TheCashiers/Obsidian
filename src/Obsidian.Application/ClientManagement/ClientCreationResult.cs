using System;

namespace Obsidian.Application.ClientManagement
{
    public class ClientCreationResult
    {
        public bool Succeed { get; set; }
        public string Message { get; set; }
        public Guid Id { get; set; }
    }
}