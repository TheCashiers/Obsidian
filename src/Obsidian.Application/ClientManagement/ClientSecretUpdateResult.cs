using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Obsidian.Application.ClientManagement
{
    public class ClientSecretUpdateResult
    {
        public bool Succeed { get; set; }
        public string Message { get; set; }
        public string Secret { get; set; }
    }
}
