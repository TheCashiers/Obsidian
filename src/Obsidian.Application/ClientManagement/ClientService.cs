using Obsidian.Domain;
using Obsidian.Domain.Repositories;
using Obsidian.Foundation;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace Obsidian.Application.ClientManagement
{
    public class ClientService
    {
        private readonly IClientRepository _repo;
        public ClientService(IClientRepository repo)
        {
            _repo = repo;
        }

        public async Task<Client> CreateClient(string displayName,string redirectUri)
        {
            var client = Client.Create(Guid.NewGuid(), redirectUri);
            client.DisplayName = displayName;
            client.UpdateSecret();
            await _repo.AddAsync(client);
            return client;
        }

        public async Task<Client> UpdateClientSecret(Guid clientId)
        {
            var client = await _repo.FindByIdAsync(clientId);
            if (client == null) throw new EntityNotFoundException($"Can not find client with id {clientId}");
            client.UpdateSecret();
            await _repo.SaveAsync(client);
            return client;
        }

        public async Task<Client> UpdateClient(Guid clientId,string displayName,string redirectUri)
        {
            var client = await _repo.FindByIdAsync(clientId);
            if (client == null) throw new EntityNotFoundException($"Can not find client with id {clientId}");
            client.DisplayName = displayName;
            client.RedirectUri = new Uri(redirectUri);
            await _repo.SaveAsync(client);
            return client;
        }
}
}
