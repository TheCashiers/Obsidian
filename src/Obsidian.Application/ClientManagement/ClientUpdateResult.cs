using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Obsidian.Application.ClientManagement
{
    public class ClientUpdateResult
    {
        public bool Succeed { get; set; }
        public string Message { get; set; }
    }
}
