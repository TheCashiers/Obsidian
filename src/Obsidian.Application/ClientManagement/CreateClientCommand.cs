using Obsidian.Foundation.ProcessManagement;

namespace Obsidian.Application.ClientManagement
{
    public class CreateClientCommand : Command<ClientCreationResult>
    {
        public string DisplayName { get; set; }
        public string RedirectUri { get; set; }

    }
}