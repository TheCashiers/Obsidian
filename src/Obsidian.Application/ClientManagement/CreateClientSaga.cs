using Obsidian.Application.ProcessManagement;
using Obsidian.Domain.Repositories;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Obsidian.Domain;
namespace Obsidian.Application.ClientManagement
{
    public class CreateClientSaga : Saga, IStartsWith<CreateClientCommand, ClientCreationResult>
    {
        private bool _isCompleted;
        private readonly IClientRepository _repo;

        public CreateClientSaga(IClientRepository repo)
        {
            _repo = repo;
        }

        public async Task<ClientCreationResult> StartAsync(CreateClientCommand command)
        {
            _isCompleted = true;
            var client = Client.Create(Guid.NewGuid(), command.RedirectUri);
            client.UpdateSecret();
            await _repo.AddAsync(client);
            return new ClientCreationResult
            {
                Succeed = true,
                Message = $"Client successfully created.",
                Id = client.Id
            };
        }

        protected override bool IsProcessCompleted() => _isCompleted;
    }
}
