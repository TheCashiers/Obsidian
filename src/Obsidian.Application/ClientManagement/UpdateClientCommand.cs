using Obsidian.Application.ProcessManagement;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Obsidian.Application.ClientManagement
{
    public class UpdateClientCommand : Command<Result<UpdateClientCommand>>
    {
        public Guid ClientId { get; set; }
        public string DisplayName { get; set; }
        public string RedirectUri { get; set; }
    }
}
