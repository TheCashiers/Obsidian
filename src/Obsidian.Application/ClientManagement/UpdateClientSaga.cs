using Obsidian.Application.ProcessManagement;
using Obsidian.Domain.Repositories;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Obsidian.Application.ClientManagement
{
    public class UpdateClientSaga : Saga, IStartsWith<UpdateClientCommand, MessageResult>
                                        , IStartsWith<UpdateClientSecretCommand, ClientSecretUpdateResult>

    {

        private bool _isCompleted;
        private readonly IClientRepository _repo;

        public UpdateClientSaga(IClientRepository repo)
        {
            _repo = repo;
        }

        public async Task<MessageResult> StartAsync(UpdateClientCommand command)
        {
            _isCompleted = true;
            var client = await _repo.FindByIdAsync(command.ClientId);
            //not exist
            if (client == null)
            {
                return new MessageResult
                {
                    Succeed = false,
                    Message = $"Client of Client Id {command.ClientId} doesn't exist."
                };
            }
            //update client
            client.DisplayName = command.DisplayName;
            client.RedirectUri = new Uri(command.RedirectUri);
            await _repo.SaveAsync(client);
            return new MessageResult
            {
                Succeed =true,
                Message = $"Info of Client {client.Id} changed."
            };
        }

        public async Task<ClientSecretUpdateResult> StartAsync(UpdateClientSecretCommand command)
        {
            _isCompleted = true;
            var client = await _repo.FindByIdAsync(command.ClientId);
            //doesn't exist
            if (client == null)
                return new ClientSecretUpdateResult
                {
                    Succeed = false,
                    Message = $"Client of Client Id {command.ClientId} doesn't exist."
                };
            //update
            client.UpdateSecret();
            await _repo.SaveAsync(client);
            return new ClientSecretUpdateResult
            {
                Succeed = true,
                Message = $"Secret of Client {client.Id} changed.",
                Secret = client.Secret
            };
        }

        protected override bool IsProcessCompleted() => _isCompleted;

    }
}
