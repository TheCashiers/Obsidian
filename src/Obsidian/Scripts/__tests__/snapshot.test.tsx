import * as React from "react";
import { create } from "react-test-renderer";
import { ClientList } from "../components/ClientManagement";
import { MaterialForm, MaterialInput } from "../components/Form";
import { List } from "../components/List";
import { NotificationCenter } from "../components/Notification";
import { PortalHeader, UserInfo } from "../components/PortalElements";
import { PortalIndex } from "../components/PortalIndex";
import { ScopeList } from "../components/ScopeManagement";
import { UserList } from "../components/UserManagement";
import { INamedEvent } from "../containers/FormContainer";

const mockChange = jest.fn<INamedEvent>();
const mockSubmit = jest.fn();
const mockDismiss = jest.fn();

it("List render snapshot", () => {
    const list = create(
        <List
            items={[
            { userName: "foo", id: "bar" },
            { displayName: "foo1", id: "bar1" },
            { scopeName: "foo2", id: "bar2" },
            ]}
        />,
    ).toJSON();
    expect(list).toMatchSnapshot();
});

it("ClientList render snapshot", () => {
    const clientList = create(
        <ClientList
            clients={[
            { displayName: "foo", id: "bar" },
            { displayName: "foo1", id: "bar1" },
            ]}
        />,
    ).toJSON();
    expect(clientList).toMatchSnapshot();
});

it("MaterialForm render snapshot", () => {
    const materialForm = create(
        <MaterialForm action="Create sth" origin="../foo" onSubmit={mockSubmit}>
            <MaterialInput
                name="username"
                label="Username"
                onInputChange={mockChange}
                value="foo"
                placeholder="Username..."
                type="text"
            />
            <MaterialInput
                name="redirectUri"
                label="Redirect Uri"
                onInputChange={mockChange}
                value="http://bar.com"
                placeholder="http://example.com"
                type="text"
            />
        </MaterialForm>,
    ).toJSON();
    expect(materialForm).toMatchSnapshot();
});

it("PortalHeader render snapshot", () => {
    const portalHeader = create(
        <PortalHeader token="MockedToken" filter="" handleFilterChange={mockChange} />,
    ).toJSON();
    expect(portalHeader).toMatchSnapshot();
});

it("PortalIndex render snapshot", () => {
    const portalIndex = create(
        <PortalIndex />,
    ).toJSON();
    expect(portalIndex).toMatchSnapshot();
});

it("ScopeList render snapshot", () => {
    const scopeList = create(
        <ScopeList
            scopes={[
            { displayName: "foo", id: "bar" },
            { displayName: "foo1", id: "bar1" },
            ]}
        />,
    ).toJSON();
    expect(scopeList).toMatchSnapshot();
});

it("UserList render snapshot", () => {
    const userList = create(
        <UserList
            users={[
            { displayName: "foo", id: "bar" },
            { displayName: "foo1", id: "bar1" },
            ]}
        />,
    ).toJSON();
    expect(userList).toMatchSnapshot();
});

it("Notifications render snapshot", () => {
    const notification = create(
        <NotificationCenter
            items={[
            { info: "foo", state: 1 },
            { info: "bar", state: 2 },
            ]}
            shouldShow={true}
            handleDismiss={mockDismiss}
        />,
    ).toJSON();
    expect(notification).toMatchSnapshot();
});
