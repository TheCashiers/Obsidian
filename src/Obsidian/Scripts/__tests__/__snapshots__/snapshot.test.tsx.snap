// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientList render snapshot 1`] = `
<div
  className="content-wrapper content"
>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="btn btn-lg btn-success"
    >
      Create Client
      ...
    </button>
  </a>
  <div
    className="list-group"
  >
    <div>
      <div
        className="list-group-item"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <div
          className="row-action-primary"
        >
          <i
            className="fa fa-server fa-2x"
          />
        </div>
        <div
          className="row-content"
        >
          <div
            className="least-content"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-lg btn-primary btn-raised"
              >
                Edit
              </button>
            </a>
          </div>
          <h4
            className="list-group-item-heading"
          >
            foo
          </h4>
          <p
            className="list-group-item-text"
          >
            bar
          </p>
        </div>
      </div>
      <div
        className="list-group-separator"
      />
    </div>
    <div>
      <div
        className="list-group-item"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <div
          className="row-action-primary"
        >
          <i
            className="fa fa-server fa-2x"
          />
        </div>
        <div
          className="row-content"
        >
          <div
            className="least-content"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-lg btn-primary btn-raised"
              >
                Edit
              </button>
            </a>
          </div>
          <h4
            className="list-group-item-heading"
          >
            foo1
          </h4>
          <p
            className="list-group-item-text"
          >
            bar1
          </p>
        </div>
      </div>
      <div
        className="list-group-separator"
      />
    </div>
  </div>
</div>
`;

exports[`List render snapshot 1`] = `
<div
  className="content-wrapper content"
>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="btn btn-lg btn-success"
    >
      ...
    </button>
  </a>
  <div
    className="list-group"
  >
    <div>
      <div
        className="list-group-item"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <div
          className="row-action-primary"
        >
          <i
            className="fa undefined fa-2x"
          />
        </div>
        <div
          className="row-content"
        >
          <div
            className="least-content"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-lg btn-primary btn-raised"
              >
                Edit
              </button>
            </a>
          </div>
          <h4
            className="list-group-item-heading"
          >
            foo
          </h4>
          <p
            className="list-group-item-text"
          >
            bar
          </p>
        </div>
      </div>
      <div
        className="list-group-separator"
      />
    </div>
    <div>
      <div
        className="list-group-item"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <div
          className="row-action-primary"
        >
          <i
            className="fa undefined fa-2x"
          />
        </div>
        <div
          className="row-content"
        >
          <div
            className="least-content"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-lg btn-primary btn-raised"
              >
                Edit
              </button>
            </a>
          </div>
          <h4
            className="list-group-item-heading"
          >
            foo1
          </h4>
          <p
            className="list-group-item-text"
          >
            bar1
          </p>
        </div>
      </div>
      <div
        className="list-group-separator"
      />
    </div>
    <div>
      <div
        className="list-group-item"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <div
          className="row-action-primary"
        >
          <i
            className="fa undefined fa-2x"
          />
        </div>
        <div
          className="row-content"
        >
          <div
            className="least-content"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-lg btn-primary btn-raised"
              >
                Edit
              </button>
            </a>
          </div>
          <h4
            className="list-group-item-heading"
          >
            foo2
          </h4>
          <p
            className="list-group-item-text"
          >
            bar2
          </p>
        </div>
      </div>
      <div
        className="list-group-separator"
      />
    </div>
  </div>
</div>
`;

exports[`MaterialForm render snapshot 1`] = `
<div
  className="well bs-component col-md-6"
>
  <form
    className="form-horizontal"
    onSubmit={undefined}
  >
    <fieldset>
      <legend>
        Create sth
      </legend>
    </fieldset>
    <div
      className="form-group"
    >
      <label
        className="col-md-2 control-label"
      >
        Username
      </label>
      <div
        className="col-md-10"
      >
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          placeholder="Username..."
          required={true}
          type="text"
          value="foo"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <label
        className="col-md-2 control-label"
      >
        Redirect Uri
      </label>
      <div
        className="col-md-10"
      >
        <input
          className="form-control"
          name="redirectUri"
          onChange={[Function]}
          placeholder="http://example.com"
          required={true}
          type="text"
          value="http://bar.com"
        />
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="col-md-10 col-md-offset-2"
      >
        <a
          onClick={[Function]}
          style={Object {}}
        >
          <button
            className="btn btn-default"
            type="button"
          >
            Cancel
          </button>
        </a>
        <button
          className="btn btn-primary"
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Notifications render snapshot 1`] = `
<div
  style={
    Object {
      "bottom": -120,
      "left": "20px",
      "opacity": 0,
      "position": "fixed",
      "zIndex": 99999,
    }
  }
>
  <div
    className="alert alert-danger"
    onClick={undefined}
    style={
      Object {
        "borderRadius": "4px",
        "clear": "both",
        "color": "rgba(255,255,255, 0.84)",
        "cursor": "pointer",
        "fontSize": "14px",
        "marginBottom": "10px",
        "maxWidth": "568px",
        "minWidth": "288px",
        "overflow": "hidden",
        "padding": "15px",
      }
    }
  >
    <span>
      foo
    </span>
  </div>
  <div
    className="alert alert-warning"
    onClick={undefined}
    style={
      Object {
        "borderRadius": "4px",
        "clear": "both",
        "color": "rgba(255,255,255, 0.84)",
        "cursor": "pointer",
        "fontSize": "14px",
        "marginBottom": "10px",
        "maxWidth": "568px",
        "minWidth": "288px",
        "overflow": "hidden",
        "padding": "15px",
      }
    }
  >
    <span>
      bar
    </span>
  </div>
</div>
`;

exports[`PortalHeader render snapshot 1`] = `
<header
  className="main-header"
>
  <nav
    className="navbar navbar-static-top"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          onClick={[Function]}
          style={Object {}}
        >
          Obsidian
        </a>
        <button
          className="navbar-toggle collapsed"
          data-target="#navbar-collapse"
          data-toggle="collapse"
          type="button"
        >
          <i
            className="fa fa-bars"
          />
        </button>
      </div>
      <div
        className="collapse navbar-collapse"
        id="navbar-collapse"
      >
        <ul
          className="nav navbar-nav"
        >
          <li>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Users
            </a>
          </li>
          <li>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Clients
            </a>
          </li>
          <li>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Scopes
            </a>
          </li>
          <li
            className="dropdown"
          >
            <a
              className="dropdown-toggle"
              data-toggle="dropdown"
              href="#"
            >
              Dropdown
              <span
                className="caret"
              />
            </a>
            <ul
              className="dropdown-menu"
              role="menu"
            >
              <li>
                <a
                  href="#"
                >
                  Action
                </a>
              </li>
              <li>
                <a
                  href="#"
                >
                  Another action
                </a>
              </li>
              <li>
                <a
                  href="#"
                >
                  Something else here
                </a>
              </li>
              <li
                className="divider"
              />
              <li>
                <a
                  href="#"
                >
                  Separated link
                </a>
              </li>
              <li
                className="divider"
              />
              <li>
                <a
                  href="#"
                >
                  One more separated link
                </a>
              </li>
            </ul>
          </li>
        </ul>
        <form
          className="navbar-form navbar-left"
          role="search"
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="navbar-search-input"
              name="filter"
              onChange={undefined}
              placeholder="Search"
              type="text"
              value={undefined}
            />
          </div>
        </form>
        <ul
          className="nav navbar-nav navbar-right"
        >
          <li
            className="dropdown user user-menu"
          >
            <a
              className="dropdown-toggle"
              data-toggle="dropdown"
              href="#"
            >
              <span
                className="hidden-xs"
              >
                Henry Chu
              </span>
            </a>
            <ul
              className="dropdown-menu "
            >
              <li
                style={
                  Object {
                    "background": "#EEE",
                    "color": "black",
                    "height": "200px",
                    "textAlign": "center",
                    "verticalAlign": "center",
                    "width": "400px",
                  }
                }
              >
                <h2>
                  Henry Chu
                   - 
                  Administrator
                  <br />
                  <small>
                    A real boss.
                  </small>
                </h2>
              </li>
              <li
                className="user-footer"
              >
                <div
                  className="pull-left"
                >
                  <a
                    className="btn btn-default btn-flat"
                    href="#"
                  >
                    Profile
                  </a>
                </div>
                <div
                  className="pull-right"
                >
                  <button
                    className="btn btn-default btn-flat"
                    onClick={[Function]}
                  >
                    Sign out
                  </button>
                </div>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</header>
`;

exports[`PortalIndex render snapshot 1`] = `
<div
  className="content-wrapper"
  style={
    Object {
      "margin": "30px",
    }
  }
>
  <h1>
    Welcome to Obsidian Management Portal!
  </h1>
  <br />
  <h2>
    You can manage all the users, claims and clients here.
  </h2>
  <h2>
    Select what you need in the navigation bar.
  </h2>
</div>
`;

exports[`ScopeList render snapshot 1`] = `
<div
  className="content-wrapper content"
>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="btn btn-lg btn-success"
    >
      Create Scope
      ...
    </button>
  </a>
  <div
    className="list-group"
  >
    <div>
      <div
        className="list-group-item"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <div
          className="row-action-primary"
        >
          <i
            className="fa fa-puzzle-piece fa-2x"
          />
        </div>
        <div
          className="row-content"
        >
          <div
            className="least-content"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-lg btn-primary btn-raised"
              >
                Edit
              </button>
            </a>
          </div>
          <h4
            className="list-group-item-heading"
          >
            foo
          </h4>
          <p
            className="list-group-item-text"
          >
            bar
          </p>
        </div>
      </div>
      <div
        className="list-group-separator"
      />
    </div>
    <div>
      <div
        className="list-group-item"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <div
          className="row-action-primary"
        >
          <i
            className="fa fa-puzzle-piece fa-2x"
          />
        </div>
        <div
          className="row-content"
        >
          <div
            className="least-content"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-lg btn-primary btn-raised"
              >
                Edit
              </button>
            </a>
          </div>
          <h4
            className="list-group-item-heading"
          >
            foo1
          </h4>
          <p
            className="list-group-item-text"
          >
            bar1
          </p>
        </div>
      </div>
      <div
        className="list-group-separator"
      />
    </div>
  </div>
</div>
`;

exports[`UserList render snapshot 1`] = `
<div
  className="content-wrapper content"
>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="btn btn-lg btn-success"
    >
      Create User
      ...
    </button>
  </a>
  <div
    className="list-group"
  >
    <div>
      <div
        className="list-group-item"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <div
          className="row-action-primary"
        >
          <i
            className="fa fa-user fa-2x"
          />
        </div>
        <div
          className="row-content"
        >
          <div
            className="least-content"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-lg btn-primary btn-raised"
              >
                Edit
              </button>
            </a>
          </div>
          <h4
            className="list-group-item-heading"
          >
            foo
          </h4>
          <p
            className="list-group-item-text"
          >
            bar
          </p>
        </div>
      </div>
      <div
        className="list-group-separator"
      />
    </div>
    <div>
      <div
        className="list-group-item"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        <div
          className="row-action-primary"
        >
          <i
            className="fa fa-user fa-2x"
          />
        </div>
        <div
          className="row-content"
        >
          <div
            className="least-content"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <button
                className="btn btn-lg btn-primary btn-raised"
              >
                Edit
              </button>
            </a>
          </div>
          <h4
            className="list-group-item-heading"
          >
            foo1
          </h4>
          <p
            className="list-group-item-text"
          >
            bar1
          </p>
        </div>
      </div>
      <div
        className="list-group-separator"
      />
    </div>
  </div>
</div>
`;
